src/Database/Driver/sparql/Connection.php

```diff
@@ -217,6 +214,54 @@ public function getConnectionOptions(): array {
    */
   public function destroy():void {}
 
+  /**
+   * Logs a query duration in the DB logger.
+   *
+   * @param string $query
+   *   The query to be logged.
+   * @param array $args
+   *   Arguments passed to the query.
+   * @param float $duration
+   *   The duration of the query run.
+   *
+   * @throws \RuntimeException
+   *   If an attempt to log was made but the logger is not started.
+   */
+  protected function log(string $query, array $args, float $duration): void {
+    if (!$this->logger) {
+      throw new \RuntimeException('Cannot log query as the logger is not started.');
+    }
+    $this->logger->log($this->getStatement()->setQuery($query), $args, $duration);
+  }
+
+  /**
+   * Returns and statically caches a DB statement stub used to log a query.
+   *
+   * The Drupal core database logger cannot be swapped because, instead of being
+   * injected, is hardcoded in \Drupal\Core\Database\Database::startLog(). But
+   * the \Drupal\Core\Database\Log::log() is expecting a database statement of
+   * type \Drupal\Core\Database\StatementInterface as first argument and the
+   * SPARQL database driver uses no StatementInterface class. We workaround this
+   * limitation by faking a database statement object just to honour the logger
+   * requirement. We use a statement stub that only stores the connection and
+   * the query to be used when logging the event.
```

Comment:
Good idea to have this disclaimer in both the calling code and the class itself.